modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/enricher/DynamicMetadataModelEnricher.java

```diff
@@ -96,20 +93,13 @@ private void enrichParameter(ParameterDeclaration declaration) {
     }
   }
 
-  private void enrichSourceMetadata(SourceDeclaration sourceDeclaration) {
-    final Optional<ImplementingTypeModelProperty> modelProperty =
-        sourceDeclaration.getModelProperty(ImplementingTypeModelProperty.class);
-    if (modelProperty.isPresent()) {
-      final Class<?> operationMethod = modelProperty.get().getType();
-      MetadataScopeAdapter metadataScope = new MetadataScopeAdapter(getMetadataScope(operationMethod));
-      MetadataResolverFactory metadataResolverFactory = getMetadataResolverFactory(metadataScope);
-      sourceDeclaration.setMetadataResolverFactory(metadataResolverFactory);
-
-      if (metadataScope.isCustomScope()) {
-        declareOutput(sourceDeclaration.getOutput(), sourceDeclaration.getOutputAttributes(), metadataScope);
-        declareComponentMetadataKeyId(sourceDeclaration);
-      }
-    }
+  private void enrichSourceMetadata(SourceDeclaration declaration) {
+    declaration.getModelProperty(ImplementingTypeModelProperty.class)
+        .ifPresent(prop -> {
+          final Class<?> sourceType = prop.getType();
+          MetadataScopeAdapter metadataScope = new MetadataScopeAdapter(extensionType, sourceType);
```

Comment:
there's some sense to this comment but it isn't completely accurate. Let's revisit this when we do the sources rework